app/src/main/java/org/jellyfin/mobile/player/source/MediaSourceResolver.kt

```diff
@@ -36,10 +37,30 @@ class MediaSourceResolver(
                     maxStreamingBitrate = /* 1 GB/s */ 1_000_000_000,
                 ),
             )
-
             response.mediaSources?.let { sources ->
                 sources.find { source -> source.id?.toUUIDOrNull() == itemId } ?: sources.firstOrNull()
-            } ?: return Result.failure(PlayerException.UnsupportedContent())
+            }.let { source ->
+                when(source?.requiresOpening) {
+                    false -> source
+                    true -> {
+                        val liveStreamResponse by mediaInfoApi.openLiveStream(
+                            data = OpenLiveStreamDto(
+                                openToken = source.openToken,
+                                itemId = itemId,
+                                userId = apiClient.userId,
+                                playSessionId = response.playSessionId,
+                                deviceProfile = deviceProfile,
+                                startTimeTicks = startTimeTicks,
+                                audioStreamIndex = audioStreamIndex,
+                                subtitleStreamIndex = subtitleStreamIndex,
+                                maxStreamingBitrate = /* 1 GB/s */ 1_000_000_000,
+                            )
+                        )
+                        liveStreamResponse.mediaSource ?: return Result.failure(PlayerException.UnsupportedContent())
+                    }
+                    null -> return Result.failure(PlayerException.UnsupportedContent())
+                }
+            }
```

Comment:
I'd replace the let call with an extra variable and use an if instead of when (just keep the null check as is).

Like so:
```kotlin
            val source = response.mediaSources?.let { sources ->
                sources.find { source -> source.id?.toUUIDOrNull() == itemId } ?: sources.firstOrNull()
            } ?: return Result.failure(PlayerException.UnsupportedContent())

            if (source.requiresOpening) {
                val liveStreamResponse by mediaInfoApi.openLiveStream(
                     data = OpenLiveStreamDto(
                         openToken = source.openToken,
                         itemId = itemId,
                         userId = apiClient.userId,
                         playSessionId = response.playSessionId,
                         deviceProfile = deviceProfile,
                         startTimeTicks = startTimeTicks,
                         audioStreamIndex = audioStreamIndex,
                         subtitleStreamIndex = subtitleStreamIndex,
                         maxStreamingBitrate = /* 1 GB/s */ 1_000_000_000,
                     )
                )

                liveStreamResponse.mediaSource ?: Result.failure(PlayerException.UnsupportedContent())
            } else {
                source
            }
```

Maybe even extract the code in the if block into another (private) function.

Reply:
Good call on suggestions. I will make the edits and take a look at closing the stream this evening.

Reply:
Regarding the closing: you could add a `requiresClosing` boolean flag to the constructor of the JellyfinMediaSource and process it in the PlayerViewModel.

The open/close methods could maybe be moved into the source class too, not sure if that works though.

Reply:
I did some more investigation this evening, and I found a more elegant solution. Adding `autoOpenLiveStream = true` to the PlaybackInfoDto for the original playback info request accomplishes the same thing server side (see: [MediaInfoController.cs](https://github.com/jellyfin/jellyfin/blob/ea7c7ea48c49750c182bf1f2c9a2496c5297aa8f/Jellyfin.Api/Controllers/MediaInfoController.cs#L200)). Additionally, while reading through the server code, it seems like the server tracks how many consumers are active and closes the live stream itself when there are no more active players. I confirmed via logs that this occurs as expected.

The only downside I could see with this approach would be if the app was resolving media sources well before playing them (for example, when adding them to a queue). However, I found that even when items are queued, the media is not resolved until they become active.

I updated my branch to reflect the changes above. I've tested them locally and they work for me.